### ISSUE TITLE
command "top_chunk" doesn't work properly

### ISSUE THREAD

**Issue by Yuuoniy** [pwndbg/pwndbg#688]
### Description
Hi, developers:
      I found that command "top_chunk" sometimes fails to print the top_chunk correctly. 
and it doesn't match with the top_chunk the "heap" command print. there might be a bug in command "top_chunk"
there is a case I came acorss:
"heap" commad shows the chunks:
```
pwndbg> heap
0x19c4000 PREV_INUSE {
  prev_size = 0, 
  size = 145, 
  fd = 0x6161616161616161, 
  bk = 0x20ff1, 
  fd_nextsize = 0x602108, 
  bk_nextsize = 0x602110
}
0x19c4090 FASTBIN {
  prev_size = 0, 
  size = 33, 
  fd = 0x6161616161616161, 
  bk = 0x6161616161616161, 
  fd_nextsize = 0xa0, 
  bk_nextsize = 0x90
}
0x19c40b0 {
  prev_size = 160, 
  size = 144, 
  fd = 0x6262626262626200, 
  bk = 0x6262626262626262, 
  fd_nextsize = 0x0, 
  bk_nextsize = 0x0
}
0x19c4140 PREV_INUSE {
  prev_size = 0, 
  size = 134849, 
  fd = 0x0, 
  bk = 0x0, 
  fd_nextsize = 0x0, 
  bk_nextsize = 0x0
}
```
while "top_chunk" return a fake top_chunk: 
```
pwndbg> top_chunk
0x19c4010 PREV_INUSE {
  prev_size = 7016996765293437281, 
  size = 135153, 
  fd = 0x602108, 
  bk = 0x602110, 
  fd_nextsize = 0x6262626262626262, 
  bk_nextsize = 0x6262626262626262
}
```
and the memory layout is :
```
pwndbg> x/16gx 0x19c4000 
0x19c4000:	0x0000000000000000	0x0000000000000091
0x19c4010:	0x6161616161616161	0x0000000000020ff1
0x19c4020:	0x0000000000602108	0x0000000000602110
0x19c4030:	0x6262626262626262	0x6262626262626262
0x19c4040:	0x6262626262626262	0x6262626262626262
0x19c4050:	0x6262626262626262	0x6262626262626262
0x19c4060:	0x6262626262626262	0x6262626262626262
0x19c4070:	0x0000000000000060	0x0000000000000000
pwndbg> 
0x19c4080:	0x0000000000000000	0x0000000000000000
0x19c4090:	0x0000000000000000	0x0000000000000021
0x19c40a0:	0x6161616161616161	0x6161616161616161
0x19c40b0:	0x00000000000000a0	0x0000000000000090
0x19c40c0:	0x6262626262626200	0x6262626262626262
0x19c40d0:	0x0000000000000000	0x0000000000000000
0x19c40e0:	0x0000000000000000	0x0000000000000000
0x19c40f0:	0x0000000000000000	0x0000000000000000
pwndbg> 
0x19c4100:	0x0000000000000000	0x0000000000000000
0x19c4110:	0x0000000000000000	0x0000000000000000
0x19c4120:	0x0000000000000000	0x0000000000000000
0x19c4130:	0x0000000000000000	0x0000000000000000
0x19c4140:	0x0000000000000000	0x0000000000020ec1
0x19c4150:	0x0000000000000000	0x0000000000000000
0x19c4160:	0x0000000000000000	0x0000000000000000
0x19c4170:	0x0000000000000000	0x0000000000000000
```

As a reslut, command "vis_heap_chunks" doesn't work as well.
```
pwndbg> vis

0x19c4000	0x0000000000000000	0x0000000000000091	................
0x19c4010	0x6161616161616161	0x0000000000020ff1	aaaaaaaa........ <-- Top chunk
```
obviously both command "top_chunk" and "vis" print a wrong reslut. top_chunk is in `0x19c4140` rather than `0x19c4010`

### Steps to reproduce
If you want to reproduce
here is the binary and exp I use
https://github.com/ctf-wiki/ctf-challenges/tree/master/pwn/heap/unlink/2016_zctf_note2

### My setup
Gdb:      8.2
Python:   3.5.2 (default, Oct  8 2019, 13:06:37)  [GCC 5.4.0 20160609]
Pwndbg:   1.1.0 build: 2f69ddc

**Comment by Yuuoniy** [comment]
well, I've dived into this, found out that it's due to the symbol main_arena, `main_arena['top'] ` sometimes return a fake top_chunk address.
I looked into the implementation of top_chunk command, and found bugs:
1.
https://github.com/pwndbg/pwndbg/blob/2f69ddce653c7fbabe9effcd42bf452c57b15fab/pwndbg/commands/heap.py#L223-L224 
'Page' object has no attribute 'size', you can use memsz instead
2.
https://github.com/pwndbg/pwndbg/blob/2f69ddce653c7fbabe9effcd42bf452c57b15fab/pwndbg/commands/heap.py#L238-L240
It doesn't make sense addr+= size twice.

and not get top_chunk address by `main_arena['top'] ` .
I've fixed these bugs in my local version, and both top_chunk and vis_heap_chunks work now.

**Comment by disconnect3d** [comment]
Fixed in #691, thanks!
